cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ * Copyright (c) OmniSci, Inc. and its affiliates.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef JITLIB_LLVM_JIT_VALUE_H
+#define JITLIB_LLVM_JIT_VALUE_H
+
+#include "exec/nextgen/jitlib/base/Values.h"
+#include "exec/nextgen/jitlib/llvmjit/LLVMJITFunction.h"
+#include "exec/nextgen/jitlib/llvmjit/LLVMJITUtils.h"
+
+namespace jitlib {
+
+template <TypeTag Type>
+class Value<Type, LLVMJITFunction> final
```

Comment:
any chance we need another function type besides LLVMJITFunction? if not, the template argument for function type should be removed.

Reply:
This is TBD, whether need to support other backend.